### DISCUSSION TITLE
Preventing the user to provide extra body param (without using a pydantic model)

### DISCUSSION THREAD

**Discussion by Thytu** [fastapi/fastapi#9849] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post(
    ENDPOINT_PATH,
)
async def some_endpoint(
    some_param: some_func() = Form(some_other_func())
) -> Dict:
    pass
```

### Description

I would like to know if there is currently anyway of preventing the user to provides extra params to the body without using `pydantic` model as some of my params are files and other are `Form` type.

Is there currently anyway of doing that?

Side note: I tried using `**kwargs` but it seems to break the swagger and we can't set it as an optional argument.

### Operating System

Linux

### FastAPI Version

0.97.0

### Python Version

3.9.14

**Comment by YuriiMotov** [comment, marked as answer]
No, there is no way to achieve this without Pydantic model.

Since FastAPI 0.113.0 you can use models to define Form fields and you can specify `extra="forbid"` to restric clients from passing extra fields.

Docs: https://fastapi.tiangolo.com/tutorial/request-form-models/#forbid-extra-form-fields
